client-core/src/transaction_builder/unauthorized_wallet_transaction_builder.rs

```diff
@@ -29,4 +30,8 @@ impl WalletTransactionBuilder for UnauthorizedWalletTransactionBuilder {
     fn obfuscate(&self, _: SignedTransaction) -> Result<TxAux> {
         Err(ErrorKind::PermissionDenied.into())
     }
+
+    fn decrypt_tx(&self, _txid: TxId, _private_key: &PrivateKey) -> Result<Transaction> {
```

Comment:
The only two methods of trait `TransactionObfuscation` get re-exported by trait `WalletTransactionBuilder` ;D

Reply:
I think if I don't use the `WalletTransactionBuilder`, I have to put a `TransactionObfuscation` implied object into the `DefaultWalletClient` and  I have to deal with the `new_read_only` function where I have to create a new mock transaction obfuscation like `UnauthorizedObfuscation` and make impl the `TransactionObfuscation`, which is much expensive.